src/models.ts

```diff
@@ -27,21 +44,29 @@ export type PokemonType = {
   };
 };
 
-export const getPokemonTypeDamage = async (url: string): Promise<PokemonTypeDamage> => {
+const pokemonTypeDamageCache = new Map<string, PokemonTypeDamage>();
+
+export const fetchPokemonTypeDamage = async (
+  url: string
+): Promise<PokemonTypeDamage> => {
+  if (pokemonTypeDamageCache.has(url)) {
+    return pokemonTypeDamageCache.get(url)!;
+  }
   const response = await fetch(url);
-  const data = await response.json();
-  return data;
+  const pokemonTypeDamage = await response.json();
+  pokemonTypeDamageCache.set(url, pokemonTypeDamage);
+  return pokemonTypeDamage;
```

Comment:
feature proposal: memory cache so you can reduce HTTP calls: 
using this simple `Set`

![Captura de pantalla 2024-03-21 a las 20 48 18](https://github.com/leire16/master-inesdi-pokedex/assets/2211052/05b99f05-ac2e-4f4a-bf00-809170b7b519)